ffi/tests/generics.rs

```diff
@@ -0,0 +1,51 @@
+#![allow(unsafe_code, clippy::restriction, clippy::pedantic)]
+
+use std::{alloc::alloc, mem::MaybeUninit};
+
+use getset::Getters;
+use iroha_ffi::{ffi_export, IntoFfi, TryFromReprC};
+
+#[derive(Clone, Copy, IntoFfi, TryFromReprC, PartialEq, Eq, Debug)]
+pub struct GenericFfiStruct<T>(T);
+
+#[derive(IntoFfi, TryFromReprC, Clone, Copy, Getters)]
+#[getset(get = "pub")]
+#[ffi_export]
+pub struct FfiStruct {
+    inner: GenericFfiStruct<bool>,
+}
+
+#[ffi_export]
+pub fn freestanding(input: GenericFfiStruct<String>) -> GenericFfiStruct<String> {
+    input
+}
+
+#[test]
+fn get_return_generic() {
+    let ffi_struct = &FfiStruct {
+        inner: GenericFfiStruct(true),
+    };
+    let mut output = MaybeUninit::<*const GenericFfiStruct<bool>>::new(core::ptr::null());
+
+    unsafe {
+        FfiStruct__inner(ffi_struct.into_ffi(), output.as_mut_ptr());
+        assert_eq!(
+            TryFromReprC::try_from_repr_c(output.assume_init(), &mut ()),
+            Ok(&ffi_struct.inner)
+        );
+    }
+}
+
+#[test]
+fn freestanding_accept_and_return_generic() {
```

Comment:
while you're at it, could you add a compile test confirming that generics are not supported on freestanding functions or methods?

Reply:
if you think this is something compiler will always refuse, you could say we don't need this test